nexus/db-model/src/inventory.rs

```diff
@@ -0,0 +1,443 @@
+            nexus_inventory::CabooseWhich::SpSlot1 => CabooseWhich::SpSlot1,
+            nexus_inventory::CabooseWhich::RotSlotA => CabooseWhich::RotSlotA,
+            nexus_inventory::CabooseWhich::RotSlotB => CabooseWhich::RotSlotB,
+        }
+    }
+}
+
+impl From<CabooseWhich> for nexus_types::inventory::CabooseWhich {
+    fn from(row: CabooseWhich) -> Self {
+        use nexus_types::inventory as nexus_inventory;
+        match row {
+            CabooseWhich::SpSlot0 => nexus_inventory::CabooseWhich::SpSlot0,
+            CabooseWhich::SpSlot1 => nexus_inventory::CabooseWhich::SpSlot1,
+            CabooseWhich::RotSlotA => nexus_inventory::CabooseWhich::RotSlotA,
+            CabooseWhich::RotSlotB => nexus_inventory::CabooseWhich::RotSlotB,
+        }
+    }
+}
+
+// See [`nexus_types::inventory::SpType`].
+impl_enum_type!(
+    #[derive(SqlType, Debug, QueryId)]
+    #[diesel(postgres_type(name = "sp_type"))]
+    pub struct SpTypeEnum;
+
+    #[derive(
+        Copy,
+        Clone,
+        Debug,
+        AsExpression,
+        FromSqlRow,
+        PartialOrd,
+        Ord,
+        PartialEq,
+        Eq
+    )]
+    #[diesel(sql_type = SpTypeEnum)]
+    pub enum SpType;
+
+    // Enum values
+    Sled => b"sled"
+    Switch =>  b"switch"
+    Power => b"power"
+);
+
+impl From<nexus_types::inventory::SpType> for SpType {
+    fn from(value: nexus_types::inventory::SpType) -> Self {
+        match value {
+            nexus_types::inventory::SpType::Sled => SpType::Sled,
+            nexus_types::inventory::SpType::Power => SpType::Power,
+            nexus_types::inventory::SpType::Switch => SpType::Switch,
+        }
+    }
+}
+
+impl From<SpType> for nexus_types::inventory::SpType {
+    fn from(value: SpType) -> Self {
+        match value {
+            SpType::Sled => nexus_types::inventory::SpType::Sled,
+            SpType::Switch => nexus_types::inventory::SpType::Switch,
+            SpType::Power => nexus_types::inventory::SpType::Power,
+        }
+    }
+}
+
+/// See [`nexus_types::inventory::Collection`].
+#[derive(Queryable, Insertable, Clone, Debug, Selectable)]
+#[diesel(table_name = inv_collection)]
+pub struct InvCollection {
+    pub id: Uuid,
+    pub time_started: DateTime<Utc>,
+    pub time_done: DateTime<Utc>,
+    pub collector: String,
+}
+
+impl<'a> From<&'a Collection> for InvCollection {
+    fn from(c: &'a Collection) -> Self {
+        InvCollection {
+            id: c.id,
+            time_started: c.time_started,
+            time_done: c.time_done,
+            collector: c.collector.clone(),
+        }
+    }
+}
+
+/// See [`nexus_types::inventory::BaseboardId`].
+#[derive(Queryable, Insertable, Clone, Debug, Selectable)]
+#[diesel(table_name = hw_baseboard_id)]
+pub struct HwBaseboardId {
+    pub id: Uuid,
+    pub part_number: String,
+    pub serial_number: String,
+}
+
+impl<'a> From<&'a BaseboardId> for HwBaseboardId {
```

Comment:
Agreed.  Fixed in a55216d40.  I changed this one and the `SwCaboose` one.  I did not change the `InvCollection` one because in that case, the source object (a `Collection`) is potentially huge.